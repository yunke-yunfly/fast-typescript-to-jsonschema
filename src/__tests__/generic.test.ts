import * as path from 'path';

import genTypeSchema from '../index';

const file = path.join(__dirname, `../../jest/generic.ts`);
genTypeSchema.genJsonDataFormFile(file);

const getSchema = (type: string) => {
  const schema = genTypeSchema.getJsonSchema(file, type);
  return schema;
};

test('Generic类型_0', () => {
  expect(getSchema('Generic_3')).toMatchSnapshot();
});

test('Generic类型_1', () => {
  expect(getSchema('Generic_4')).toMatchSnapshot();
});

test('Generic类型_2', () => {
  expect(getSchema('Generic_5')).toMatchSnapshot();
});

test('Generic类型_3', () => {
  expect(getSchema('Generic_type_1')).toMatchSnapshot();
});

test('Generic默认值_类型_1', () => {
  expect(getSchema('Generic_7')).toMatchSnapshot();
});

test('Generic默认值_类型_2', () => {
  expect(getSchema('Generic_12')).toMatchSnapshot();
});

test('Generic默认值_类型_3', () => {
  expect(getSchema('Generic_13')).toMatchSnapshot();
});

test('Generic多层对象_类型_1', () => {
  expect(getSchema('Generic_9')).toMatchSnapshot();
});

test('Generic多层对象_类型_2', () => {
  expect(getSchema('Generic_11')).toMatchSnapshot();
});

test('Generic多层对象_类型_3', () => {
  expect(getSchema('Generic_13')).toMatchSnapshot();
});

test('Generic引用别的文件_类型_1', () => {
  expect(getSchema('Generic_10')).toMatchSnapshot();
});

