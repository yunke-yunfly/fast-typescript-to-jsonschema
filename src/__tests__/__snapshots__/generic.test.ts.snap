// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic多层对象_类型_1 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {},
  "properties": Object {
    "value1": Object {
      "additionalProperties": false,
      "properties": Object {
        "value2": Object {
          "additionalProperties": false,
          "properties": Object {
            "value3": Object {
              "type": "string",
            },
          },
          "required": Array [
            "value3",
          ],
          "type": "object",
        },
      },
      "required": Array [
        "value2",
      ],
      "type": "object",
    },
  },
  "required": Array [
    "value1",
  ],
  "type": "object",
}
`;

exports[`Generic多层对象_类型_2 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Generic_8<string>": Object {
      "additionalProperties": false,
      "properties": Object {
        "value1": Object {
          "additionalProperties": false,
          "properties": Object {
            "value2": Object {
              "additionalProperties": false,
              "properties": Object {
                "value3": Object {
                  "type": "string",
                },
              },
              "required": Array [
                "value3",
              ],
              "type": "object",
            },
          },
          "required": Array [
            "value2",
          ],
          "type": "object",
        },
      },
      "required": Array [
        "value1",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "value": Object {
      "items": Object {
        "$realRef": "Generic_8",
        "$ref": "#/definitions/Generic_8<string>",
      },
      "type": "array",
    },
  },
  "required": Array [
    "value",
  ],
  "type": "object",
}
`;

exports[`Generic多层对象_类型_3 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {},
  "properties": Object {
    "value": Object {
      "items": Object {
        "type": "string",
      },
      "type": "array",
    },
  },
  "required": Array [
    "value",
  ],
  "type": "object",
}
`;

exports[`Generic引用别的文件_类型_1 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {},
  "properties": Object {
    "value1": Object {
      "type": "string",
    },
    "value2": Object {
      "type": "string",
    },
  },
  "required": Array [
    "value1",
    "value2",
  ],
  "type": "object",
}
`;

exports[`Generic类型_0 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {},
  "properties": Object {
    "value1": Object {
      "type": "string",
    },
    "value2": Object {
      "type": "string",
    },
  },
  "required": Array [
    "value1",
    "value2",
  ],
  "type": "object",
}
`;

exports[`Generic类型_1 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Generic_1<number>": Object {
      "additionalProperties": false,
      "properties": Object {
        "value1": Object {
          "items": Object {
            "type": "number",
          },
          "type": "array",
        },
      },
      "required": Array [
        "value1",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "value1": Object {
      "type": "string",
    },
    "value2": Object {
      "$ref": "#/definitions/Generic_1<number>",
    },
  },
  "required": Array [
    "value1",
    "value2",
  ],
  "type": "object",
}
`;

exports[`Generic类型_2 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Generic_1<number>": Object {
      "additionalProperties": false,
      "properties": Object {
        "value1": Object {
          "items": Object {
            "type": "number",
          },
          "type": "array",
        },
      },
      "required": Array [
        "value1",
      ],
      "type": "object",
    },
    "Generic_2<Generic_1<number>>": Object {
      "additionalProperties": false,
      "properties": Object {
        "value1": Object {
          "type": "string",
        },
        "value2": Object {
          "$ref": "#/definitions/Generic_1<number>",
        },
      },
      "required": Array [
        "value1",
        "value2",
      ],
      "type": "object",
    },
    "Generic_2<string>": Object {
      "additionalProperties": false,
      "properties": Object {
        "value1": Object {
          "type": "string",
        },
        "value2": Object {
          "type": "string",
        },
      },
      "required": Array [
        "value1",
        "value2",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "value1": Object {
      "$ref": "#/definitions/Generic_1<number>",
    },
    "value2": Object {
      "$ref": "#/definitions/Generic_2<string>",
    },
    "value3": Object {
      "$ref": "#/definitions/Generic_2<string>",
    },
    "value4": Object {
      "$ref": "#/definitions/Generic_2<Generic_1<number>>",
    },
  },
  "required": Array [
    "value1",
    "value2",
    "value3",
    "value4",
  ],
  "type": "object",
}
`;

exports[`Generic类型_3 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "T": Object {},
  },
  "properties": Object {
    "value1": Object {
      "type": "string",
    },
    "value2": Object {
      "$ref": "#/definitions/T",
    },
  },
  "required": Array [
    "value1",
    "value2",
  ],
  "type": "object",
}
`;

exports[`Generic默认值_类型_1 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Generic_6<string>": Object {
      "additionalProperties": false,
      "properties": Object {
        "value": Object {
          "items": Object {
            "type": "string",
          },
          "type": "array",
        },
      },
      "required": Array [
        "value",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "value7": Object {
      "$ref": "#/definitions/Generic_6<string>",
    },
  },
  "required": Array [
    "value7",
  ],
  "type": "object",
}
`;

exports[`Generic默认值_类型_2 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Generic_8<string>": Object {},
  },
  "properties": Object {
    "value": Object {
      "items": Object {
        "$realRef": "Generic_8",
        "$ref": "#/definitions/Generic_8<string>",
      },
      "type": "array",
    },
  },
  "required": Array [
    "value",
  ],
  "type": "object",
}
`;

exports[`Generic默认值_类型_3 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {},
  "properties": Object {
    "value": Object {
      "items": Object {
        "type": "string",
      },
      "type": "array",
    },
  },
  "required": Array [
    "value",
  ],
  "type": "object",
}
`;
